bindings/python/scripts/parse.py

```diff
@@ -1,145 +1,97 @@
 
     for child in cursor.get_children():
-        if node_in_this_file(child, this_filename):
-            child_list = []
-            generate_parsed_info(
-                child, filter, lines, this_filename, depth + 1, child_list,
+        if is_node_in_this_file(cursor=child, filename=this_filename):
+            child_parsed_info = generate_parsed_info(
+                cursor=child, this_filename=this_filename, depth=depth + 1,
             )
-            if child_list and parsed_list:
-                if len(parsed_list[0]["members"]):
-                    parsed_list[0]["members"].append(child_list[0])
-                else:
-                    parsed_list[0]["members"] = child_list
+            if child_parsed_info and parsed_info:
+                parsed_info["members"].append(child_parsed_info)
 
+    return parsed_info
 
-def parse_arguments(args):
-    parser = argparse.ArgumentParser(description="C++ libclang parser")
-    parser.add_argument("files", nargs="+", help="The source files to search")
-    return parser.parse_args(args)
 
-
-def get_output_path(source, output_dir):
-    x_list = source.split("pcl/", 1)[-1]
-    x_list = x_list.split("/")
-    extra = ["pcl", "include"]
-
-    filename = x_list[-1].split(".")[0]
-    relative_dir = "/".join(x for x in x_list[:-1] if x not in extra)
-    dir = os.path.join(output_dir, relative_dir)
-
-    # ensure the new directory exists
-    if not os.path.exists(dir):
-        os.makedirs(dir)
-
-    return f"{dir}/{filename}.json"
+def get_compilation_commands(compilation_database_path, filename):
+    compilation_database = clang.CompilationDatabase.fromDirectory(
+        buildDir=compilation_database_path
+    )
+    compilation_commands = compilation_database.getCompileCommands(filename=filename)
+    # extracting argument list from the command's generator object
+    return list(compilation_commands[0].arguments)[1:-1]
 
 
 def main():
-    args = parse_arguments(sys.argv[1:])
-    index = clang.Index.create()
+    args = utils.parse_arguments(args=sys.argv[1:], script="parse")
+    for source in args.files:
+        source = utils.get_realpath(path=source)
 
-    compile_db_path = os.path.dirname("./compile_commands.json")
-    compdb = clang.CompilationDatabase.fromDirectory(compile_db_path)
+        index = clang.Index.create()
 
-    for source in args.files:
-        with open(source) as input_file:
-            lines = input_file.readlines()
+        compilation_commands = get_compilation_commands(
+            compilation_database_path="/home/divyanshu/Projects/active/pcl/bindings/python",
+            filename=source,
+        )
 
-        compile_commands = compdb.getCompileCommands(source)
-        # extracting argument list from the command's generator object
-        compile_commands = list(compile_commands[0].arguments)[1:-1]
-        tu = index.parse(source, args=compile_commands)
+        tu = index.parse(path=source, args=compilation_commands)
```

Comment:
You don't need to use `translation_unit` (and shouldn't) since the file might be a header (based on your design and usecase).

`file_ast` might be a good starting point for an alternative

Reply:
Header and impl files too can be compiled, right?
Also, this is only about the variable name, correct?

Reply:
`tu` has a very specific meaning and you're not using that context, so it's better to avoid confusion

Reply:
Good point. Changed to `source_ast`.